### ISSUE TITLE
type annotation and implementation mismatch

### ISSUE THREAD

**Issue by pinterior** [chardet/chardet#276]
`chardet.detect` (and `UniversalDetector.close`) may return `None` for `encoding` or `language` field.
but both fields are annotated as not `None` as below. so I think it's better to revert merge of #266 .

```
$ git show --oneline --no-patch
7264282 (HEAD -> main, origin/main, origin/HEAD) Add support for Python 3.11 (#274)
$ python
Python 3.8.10 (default, Jun 22 2022, 20:18:18)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import chardet
>>> chardet.__version__
'5.1.0dev0'
>>> chardet.detect(b"")
{'encoding': None, 'confidence': 0.0, 'language': None}
>>> chardet.detect(b"\x81")
{'encoding': None, 'confidence': 0.0, 'language': None}
```

https://github.com/chardet/chardet/blob/72642823f7bd4b12efe9fa066ec7a84c607e141f/chardet/__init__.py#L30-L32
https://github.com/chardet/chardet/blob/72642823f7bd4b12efe9fa066ec7a84c607e141f/chardet/resultdict.py#L10-L13

Ref: https://github.com/python/typeshed/issues/9292

**Comment by dan-blanchard** [comment]
That makes sense to me. I'll get that reverted.

**Comment by dan-blanchard** [comment]
Reverted and released in chardet 5.1.0
